lib/git.zsh

```diff
@@ -9,14 +9,27 @@ function __git_prompt_git() {
   GIT_OPTIONAL_LOCKS=0 command git "$@"
 }
 
-# Outputs current branch info in prompt format
 function git_prompt_info() {
+  # If we are on a folder not tracked by git, get out.
+  # Otherwise, check for hide-info at global and local repository level
+  if ! $(__git_prompt_git rev-parse --git-dir &> /dev/null) \
+     || [[ "$(__git_prompt_git config --get oh-my-zsh.hide-info 2>/dev/null)" == 1 ]]; then
+    return 0
+  fi
+
   local ref
-  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]]; then
-    ref=$(__git_prompt_git symbolic-ref HEAD 2> /dev/null) || \
-    ref=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null) || return 0
-    echo "$ZSH_THEME_GIT_PROMPT_PREFIX${ref#refs/heads/}$(parse_git_dirty)$ZSH_THEME_GIT_PROMPT_SUFFIX"
+  ref=$(__git_prompt_git symbolic-ref --short HEAD 2> /dev/null) \
+  || ref=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null) \
+  || return 0
+ 
+  # Use global ZSH_THEME_GIT_SHOW_UPSTREAM=1 for including upstream remote info
+  local upstream
+  if (( ${+ZSH_THEME_GIT_SHOW_UPSTREAM} )); then
+    upstream=$(__git_prompt_git rev-parse --abbrev-ref --symbolic-full-name "@{upstream}" 2>/dev/null) \
+    && upstream=" -> ${upstream}"
```

Comment:
Is the arrow ok? I used it as a first draft but maybe there's a better symbol? Maybe we should ask the users who +1'd the PR.

Reply:
I think the arrow is OK, but frankly, it duplicates content (the branch appears twice), and it is something you suggested earlier on that it should not repeat information.

Maybe go back to the `[$remote/]$branch` format?

Reply:
The upside is that the local branch may have a different name, as we also discussed before.

Reply:
> The upside is that the local branch may have a different name, as we also discussed before.

Also a branch may not be at the same commit as the upstream branch, for example:
```
$ git log --all | head -3
* 452586e7 (upstream/master, origin/master, origin/HEAD) chore: add Josh Medeski as `wp-cli` plugin maintainer
* 865d2d8a refactor(archlinux): prevent leaking local variables (#9476)
* e48be7df (master) feat(wp-cli): add aliases for DB management (#9469)
$ ZSH_THEME_GIT_SHOW_UPSTREAM=1 git_prompt_info
git:(master -> upstream/master)
```
If we only show the upstream branch, that could be misleading.

Reply:
Cool. Let's go with `$branch [-> $remote/$branch]` then 👍